decred/tests/unit/dcr/test_addrlib.py

```diff
@@ -0,0 +1,438 @@
+"""
+Copyright (c) 2019-2020, the Decred developers
+See LICENSE for details
+"""
+
+from base58 import b58decode
+
+from decred import DecredError
+from decred.crypto import crypto
+from decred.dcr import addrlib
+from decred.dcr.nets import mainnet, testnet
+from decred.util.encode import ByteArray
+
+
+def test_addresses():
+    class test:
+        def __init__(
+            self,
+            name="",
+            addr="",
+            saddr="",
+            encoded="",
+            valid=False,
+            scriptAddress=None,
+            f=None,
+            net=None,
+            skipComp=False,
+        ):
+            self.name = name
+            self.addr = addr
+            self.saddr = saddr
+            self.encoded = encoded
+            self.valid = valid
+            self.scriptAddress = scriptAddress
+            self.f = f
+            self.net = net
+            # Pubkey addresses are often printed as pkh addresses, making
+            # comparison of the decoded address impossible.
+            self.skipComp = skipComp
+
+    addrPKH = addrlib.AddressPubKeyHash
+    addrSH = addrlib.AddressScriptHash.fromScript
+    addrSHH = addrlib.AddressScriptHash
+    addrPK = addrlib.AddressSecpPubKey
+
+    tests = []
+    # Positive P2PKH tests.
+    tests.append(
```

Comment:
Why use `append` instead of a long list?

Reply:
Just translated that way. This test was originally defined in test_txscript.py. Will update to a list.